chef-config/lib/chef-config/config.rb

```diff
@@ -513,7 +513,32 @@ def self.path_accessible?(path)
     default :recipe_url, nil
 
     # Set to true if Chef is to set OpenSSL to run in FIPS mode
-    default(:fips) { ENV["CHEF_FIPS"] == "1" }
+    default(:fips) do
+      !ENV["CHEF_FIPS"].nil? || check_fips_via_ohai
+    end
+
+    # we want to synchronize this ohai call because ohai is not thread safe
+    # if this gets called in a mulithreaded context, each thread's ohai instance
+    # will call reset_system while other threads are loading plugins
+    # the destructive power of reset_system is scoped to the module and not to the instance
+    def self.check_fips_via_ohai
+      @@sync ||= Mutex.new
```

Comment:
This is just one of the weird things about everything in ruby being an object. Since a class is an instance of the `Class` class, it can have its own instance variables, which are separate from instance variables of instances of that class. E.g.,

```ruby
class AClass

  puts self.inspect

  @class_ivar = "a class instance variable"

  def self.class_ivar
    puts "class_ivar is: #{@class_ivar.inspect}"
  end

  def cant_see_class_ivar
    puts "class_ivar is: #{@class_ivar.inspect}"
  end

end

AClass.class_ivar

AClass.new.cant_see_class_ivar
```

That will output:

```
AClass
class_ivar is: "a class instance variable"
class_ivar is: nil
```

Class variables in ruby are shared by all subclasses, so there are a few times when you really need them (keeping track of plugin loading in knife is one example), but usually you want a variable scoped to just that one class, which is what class instance variables will do.